util/warnp.c

```diff
@@ -62,7 +65,18 @@ warn(const char * fmt, ...)
 		fprintf(stderr, ": %s\n", strerror(errno));
 	} else {
 		/* Print to syslog. */
-		/* Not implemented. */
+		if (fmt != NULL) {
+			/* No need to print "${name}: "; syslog does it. */
+			vsnprintf(msgbuf, WARNP_SYSLOG_MAX_LINE + 1, fmt, ap);
+			syslog(syslog_priority, "%s: %s\n", msgbuf,
+			    strerror(errno));
+		} else {
+			if (errno != 0)
```

Comment:
Why?  There's no such check in the non-syslog version.

Reply:
Oops.  Removed.